python/lsst/ts/scriptqueue/ui/request_cmd.py

```diff
@@ -0,0 +1,385 @@
+                yconfig = {}
+                for i in range(int(len(parsed.parameters)/2)):
+                    yconfig[parsed.parameters[i*2]] = parsed.parameters[i*2+1]
+                config = yaml.safe_dump(yconfig)
+                self.log.debug('Configuration: %s', config)
+            else:
+                self.log.debug("No configuration file or in line parameters.")
+
+            # Preparing to load script
+            script, is_standard = (parsed.standard, True) if parsed.standard is not None \
+                else (parsed.external, False)
+
+            # Add to the queue
+            self.log.debug(f"Adding {script} to the queue.")
+            salindex = self.model.add(script, is_standard, config)
+
+            if parsed.monitor:
+                self.monitor_script(salindex, parsed.timeout)
+            else:
+                self.log.info(f"Script index: {salindex}.")
+        except Exception as e:
+            self.log.exception(e)
+
+        return
+
+    def help_run(self):
+        """Print help for the `run` command using the command parser.
+        """
+        self.__run_parser.print_help()
+
+    def do_set_log_level(self, args):
+        """Set log level of the shell script.
+        """
+        try:
+            parsed = self.__set_log_parser.parse_args(args.split())
+            logging.basicConfig(level=parsed.vebose)
+        except SystemExit:
+            return
+
+    def help_set_log_level(self):
+        """Helper for set_log_level."""
+        self.__set_log_parser.print_help()
+
+    def do_set_script_log_level(self, args):
+        """Set log level of a script.
+        """
+        try:
+            parsed = self.__set_script_log_parser.parse_args(args.split())
+            self.log.debug(f"Setting script {parsed.index} log level to {DETAIL_LEVEL[parsed.verbose]}")
+            self.model.set_script_log_level(parsed.index,
+                                            DETAIL_LEVEL[parsed.verbose])
+        except SystemExit:
+            return
+        except Exception as e:
+            self.log.exception(e)
+
+    def help_set_script_log_level(self):
+        """Helper for set_script_log_level."""
+        self.__set_script_log_parser.print_help()
+
+    def do_set_queue_log_level(self, args):
+        """Set log level of the queue.
+        """
+        try:
+            parsed = self.__set_log_parser.parse_args(args.split())
+            self.log.debug(f"Setting queue log level to {DETAIL_LEVEL[parsed.verbose]}")
+            self.model.set_queue_log_level(DETAIL_LEVEL[parsed.verbose])
+        except SystemExit:
+            return
+        except Exception as e:
+            self.log.exception(e)
+
+    def help_set_queue_log_level(self):
+        """Helper for set_log_level."""
+        self.__set_log_parser.print_help()
+
+    def monitor_script(self, salindex, timeout):
+        """Monitor the execution of a script. Will block until
+        script is in a final state.
```

Comment:
Yes. Unfortunately, unless we go the `curses` route it does have to block. I tried some different techniques to have callbacks running in the background but none of them worked. I hate to say it but I think that, for the scope of this work, I would not like to go the `curses` way. Specially if we consider that we will have the LOVE interface to deal with most of this pretty soon. 

If eventually we find that it would be nice to have that in addition to LOVE, I would be glad to implement it with `curses`.  Hope this is fine.

Reply:
I'm puzzled by the blockage but agree with your conclusion. Maybe Cmd doesn't play well with asyncio.